[Doc] Add example to  keyboard housekeeping and some minor fixes (#19968)
qmk/qmk_firmware: docs/custom_quantum_functions.md

Similar to `matrix_scan_*`, these are called as often as the MCU can handle. To keep your board responsive, it's suggested to do as little as possible during these function calls, potentially throtting their behaviour if you do indeed require implementing something special.

### Example `void housekeeping_task_user(void)` implementation

> This example will show you how to use `void housekeeping_task_user(void)` to turn off RGB light. For RGB Matrix just use `#define RGB_MATRIX_TIMEOUT` in your `config.h`.
> 
> First add those lines to your user `config.h`.

Reviewer: ```suggestion
This example will show you how to use `void housekeeping_task_user(void)` to turn off [RGB Light](feature_rgblight.md). For RGB Matrix, the [builtin](https://docs.qmk.fm/#/feature_rgb_matrix?id=additional-configh-options) `RGB_MATRIX_TIMEOUT` should be used.

First, add the following lines to your keymap's `config.h`:
```

```c
#define RGBLIGHT_SLEEP  // enable rgblight_suspend() and rgblight_wakeup() in keymap.c
#define RGBLIGHT_TIMEOUT 900000  // ms to wait until rgblight time out, 900K ms is 15min.
```

> And add those lines to your `keymap.c`.

Reviewer: ```suggestion
Next, add the following code to your `keymap.c`:
```

### Example `void housekeeping_task_user(void)` implementation

This example will show you how to use `void housekeeping_task_user(void)` to turn off RGB light. For RGB Matrix just use `#define RGB_MATRIX_TIMEOUT` in your `config.h`.

First add those lines to your user `config.h`.

```c
#define RGBLIGHT_SLEEP  // enable rgblight_suspend() and rgblight_wakeup() in keymap.c
#define RGBLIGHT_TIMEOUT 900000  // ms to wait until rgblight time out, 900K ms is 15min.
```

And add those lines to your `keymap.c`.

```c
static uint32_t key_timer;           // timer for last keyboard activity, use 32bit value and function to make longer idle time possible
static void refresh_rgb(void);       // refreshes the activity timer and RGB, invoke whenever any activity happens
static void check_rgb_timeout(void); // checks if enough time has passed for RGB to timeout
bool is_rgb_timeout = false;         // store if RGB has timed out or not in a boolean

> void refresh_rgb()
> {

Reviewer 2: Examples should follow coding style guidelines. 

Eg: 
```suggestion
void refresh_rgb(void) {
```

Author: Just made those changes, let me know if anything else I need change.
